staging/src/k8s.io/client-go/discovery/cached/memcache.go

```diff
@@ -117,35 +119,34 @@ func (d *memCacheClient) Fresh() bool {
 func (d *memCacheClient) Invalidate() {
 	d.lock.Lock()
 	defer d.lock.Unlock()
+	d.cacheValid = false
+}
 
+// refreshLocked refreshes the state of cache. The caller must hold d.lock for
+// writing.
+func (d *memCacheClient) refreshLocked() error {
 	// TODO: Could this multiplicative set of calls be replaced by a single call
 	// to ServerResources? If it's possible for more than one resulting
 	// APIResourceList to have the same GroupVersion, the lists would need merged.
 	gl, err := d.delegate.ServerGroups()
 	if err != nil || len(gl.Groups) == 0 {
-		utilruntime.HandleError(fmt.Errorf("couldn't get current server API group list; will keep using cached value. (%v)", err))
-		return
+		return fmt.Errorf("couldn't get current server API group list; will keep using cached value. (%v)", err)
```

Comment:
This error text isn't true any more, all the calls will just fail now as the cache remains invalid. Did I miss something?

Reply:
Done.